minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromDamageImplementation.java

```diff
@@ -0,0 +1,142 @@
+package com.microsoft.Malmo.MissionHandlers;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.microsoft.Malmo.MissionHandlerInterfaces.IObservationProducer;
+import com.microsoft.Malmo.Schemas.MissionInit;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.living.LivingDeathEvent;
+import net.minecraftforge.event.entity.living.LivingHurtEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import static java.lang.Math.asin;
+import static java.lang.Math.atan2;
+
+/**
+ * Simple IObservationProducer object that pings out a whole bunch of data.<br>
+ */
+public class ObservationFromDamageImplementation extends HandlerBase implements IObservationProducer {
+    private DamageSource damageSource;
+    private float damageAmount;
+    private EntityLivingBase entity;
+    private boolean hasDied = false;
+    private boolean damageSourceReplaced = false;
+
+
+    @Override
+    public void prepare(MissionInit missionInit) {
+        MinecraftForge.EVENT_BUS.register(this);
+    }
+
+    @Override
+    public void cleanup() {
+    }
+
+    @SubscribeEvent
+    public void onDeath(LivingDeathEvent event) {
+        if (event.getEntityLiving().equals(Minecraft.getMinecraft().player)) {
+            if (this.damageSource != null) {
+                // If the agent dies from damage, a LivingHurtEvent will precede the death event. Since death events do
+                // not include the damage amount we simply update the other fields to ensure the cause of death is
+                // properly reflected
+                this.damageSourceReplaced = true;
```

Comment:
Looks like we are marking every "death from damage" event (and this will constitute a large fraction of our death events) as `stale=True` here -- is this intentional?

Reply:
> "we simply update the other fields to ensure the cause of death"

Doesn't look like you are updating other fields here -- instead you are reading from `this.damageSource` as usual?

Reply:
Hmm I want to mark something regarding the fact that the damage event is overridden in the case that we can kill the agent without damaging it and want to know if that is the case?

Reply:
I should be updating this.damageSource from event.getSource(); below

Reply:
sounds good